test/MaterializedViewsTest.cpp

```diff
@@ -454,10 +525,185 @@ TEST_F(MaterializedViewsTest, ManualConfigurations) {
   }
 }
 
+// _____________________________________________________________________________
+TEST_F(MaterializedViewsTest, serverIntegration) {
+  // Write a new materialized view using the `writeMaterializedView` method of
+  // the `Server` class.
+  {
+    // Initialize but do not start a `Server` instance on our test index.
+    Server server{4321, 1, ad_utility::MemorySize::megabytes(1), "accessToken"};
+    server.initialize(testIndexBase_, false);
+
+    ad_utility::url_parser::sparqlOperation::Query query{simpleWriteQuery_, {}};
+    ad_utility::Timer requestTimer{ad_utility::Timer::InitialStatus::Started};
+    auto cancellationHandle =
+        std::make_shared<ad_utility::CancellationHandle<>>();
+    static constexpr size_t dummyTimeLimit = 1000 * 60 * 60;  // 1 hour
+    std::chrono::milliseconds timeLimit{dummyTimeLimit};
+    server.writeMaterializedView("testViewFromServer", query, requestTimer,
+                                 cancellationHandle, timeLimit);
+  }
+
+  // Try loading the new view.
+  {
+    qlv().loadMaterializedView("testViewFromServer");
+    auto [qet, qec, parsed] = qlv().parseAndPlanQuery(
+        "SELECT * { ?s "
+        "<https://qlever.cs.uni-freiburg.de/materializedView/"
+        "testViewFromServer-o> ?o }");
+    auto res = qet->getResult(false);
+    ASSERT_TRUE(res->isFullyMaterialized());
+    EXPECT_EQ(res->idTable().numColumns(), 2);
+    EXPECT_EQ(res->idTable().numRows(), 4);
+  }
+
+  // Test the HTTP request processing of the `Server` class regarding writing
+  // and loading materialized views.
+  using ReqT = http::request<http::string_body>;
+  using ResT = std::optional<http::response<http::string_body>>;
+  auto simulateHttpRequest =
+      [&](const ReqT& request,
+          ad_utility::source_location location =
+              AD_CURRENT_SOURCE_LOC()) -> std::optional<nlohmann::json> {
+    auto l = generateLocationTrace(location);
+    boost::asio::io_context io;
+    std::future<ResT> fut = co_spawn(
+        io,
+        [](auto request, auto indexName) -> boost::asio::awaitable<ResT> {
+          // Initialize but do not start a `Server` instance on our test index.
+          Server server{4321, 1, ad_utility::MemorySize::megabytes(1),
+                        "accessToken"};
+          server.initialize(indexName, false);
+
+          // Simulate receiving the HTTP request.
+          auto result =
+              co_await server
+                  .template onlyForTestingProcess<decltype(request), ResT>(
+                      request);
+          co_return result;
+        }(request, testIndexBase_),
+        boost::asio::use_future);
+    io.run();
+    auto response = fut.get();
+    if (!response.has_value()) {
+      return std::nullopt;
+    }
```

Comment:
This very much looks like we could factor out the general infrastructure of setting up the stuff such that we can co_await some mock-return-process stuff
(In particualr the `onlyForTestingProcess` which could be used for other tests).
Then this test gets more readable + other tests get writable:)